### ISSUE TITLE
"root" environment needs a different name

### ISSUE THREAD

**Issue by ijstokes** [conda/conda#1609]
`/root` is what the lay user will think the "_root environment_" refers to.  _root_ is fairly universally assumed to mean _things associated with the root super user_.

I very strongly believe, even though it is late in the day, we need to rename _root environment_ to something like _base environment_ that will not create this confusion.

**Comment by scarescrow** [comment]
Could a feature be added to have the default root name configurable by the user in the setup?

**Comment by mingwandroid** [comment]
Personally I would oppose a configurable name for the root env as it would lead to miscommunication. Renaming it globally would do the same in the short term but would be OK in the end so I'm fine with that.

**Comment by kalefranz** [comment]
I think we should rename it.  We don't want to be adding `(root)` to the command prompt, then people thinking they're the root user.

I could go for `base`.  Are there any other ideas to consider?

**Comment by kalefranz** [comment]
When we implement this, we'll base `root` an alias for `base`.  So `source activate root` will put you right back to `base`.

**Comment by pzwang** [comment]
+1 for base.

**Comment by kalefranz** [comment]
After discussion, the consensus seems to be https://github.com/conda/conda/pull/5581 is the best solution.  We just don't show a prompt modifier when the root environment is the active environment.

The root environment is still named "root." Activate works just as it did before, i.e. when no argument is given to `source activate` or `conda activate`, then it's assumed that the root environment is the one being activated.

It was discussed that the danger with not showing a prompt modifier when the root environment is active is that users can deactivate their way out of conda not being on PATH at all.  While that's also the case currently, with no indicator of the root environment being active, the prompts will look the same whether the conda root environment is active or not.

**Comment by kalefranz** [comment]
`home` might have also been a good option to consider, in addition to `base` and `default`.

**Comment by goanpeca** [comment]
@pzwang, @teoliphant:

- I don't like the idea of not having parity between command line and GUI (Navigator), one will now be empty, the other will say root.
- `root` in general is not a good name, even if it is outside of the CLI context, it does not really provide enough meaning to understand what it is supposed to be.
- `base` or `default` make more sense, as using `source activate` with no environment activates... the "default" environment.

**Comment by kalefranz** [comment]
In the discussion yesterday, we also missed how we handle `conda info -e`, which lists all named environments.  It's actually a similar problem to what @goanpeca is confronting with Navigator.  If the root environment is now more treated as "unnamed default", then do we still call it "root" in these other UI situations.

**Comment by nehaljwani** [comment]
What problems would arise, if we just keep it one letter: `_` or `____` ? 

That way...
- We no longer need to debate what it should be 'called'. The _root_ environment still has a name (from the code's perspective), but semantics can be 'unnamed default'
- The symbol denotes the base/floor or operations
- It is not a word, but just a bunch of underscores, so maybe it strikes to users that it has a special meaning?
- This can be consistent across CLI and other UIs

**Comment by ijstokes** [comment]
"default" has some shortcomings in that it may be contextualized to mean
"the Python version in effect now". That is how we used it with the conda
environment selection extension for picking a Jupyter kernel.

I would prefer "base" or "home".

**Comment by goanpeca** [comment]
I would prefer "base" in that case, home now is too close to either the home directory in linux and windows

**Comment by kalefranz** [comment]
I created https://github.com/conda/conda/pull/5581, but it never got merged.  Which means 4.4.0rc0 was released with the root environment effectively renamed to "base" insofar as it's displayed to users in a command prompt change.  There may be other places in conda help messages, etc, where we still say "root," and we can change those in time as we come across them.

As implemented, "root" is now an alias for "base" so that `source activate`, `source activate root`, and `source activate base` are all equivalent commands.

Closing this issue now, along with #5581.
